File: src/Finos.Fdc3.Backplane.Client/InitializeParams.cs
/*
	* SPDX-License-Identifier: Apache-2.0
	* Copyright 2022 FINOS FDC3 contributors - see NOTICE file
	*/

using Finos.Fdc3.Backplane.DTO.FDC3;

namespace Finos.Fdc3.Backplane.Client
{
    /// <summary>
    /// 
    /// </summary>
    public class InitializeParams
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="desktopAgentIdentifier"></param>
        public InitializeParams(AppIdentifier desktopAgentIdentifier)
        {
            AppIdentifier = desktopAgentIdentifier;
        }

        /// <summary>
        /// App/Desktop agent identifier
        /// </summary>
        public AppIdentifier AppIdentifier { get; }

    }
}

File: src/Finos.Fdc3.Backplane.Client.JS/src/API/initializeParams.ts
/**
 * SPDX-License-Identifier: Apache-2.0
 * Copyright 2021 FINOS FDC3 contributors - see NOTICE file
 */

import { ILogger } from '@microsoft/signalr';
import { AppIdentifier } from '../DTO/MessageEnvelope';

export interface InitializeParams {
	/**
	 * App/desktop agent identifier
	 *
	 * @type {DesktopAgentIdentifier}
	 * @memberof InitializeOptions
	 */
	appIdentifier: AppIdentifier;

	/**
	 * Injected logger
	 *
	 * @type {Logger}
	 * @memberof InitializeOptions
	 */
	logger?: ILogger;

	/**
	 *
	 *
	 * @type {Url}
	 * @memberof InitializeParams
	 */
	url: string;
}
